### PR TITLE
feat: Add Plan.creators property

### PR DESCRIPTION
**Pull request by eikek**
Add a `creators` property to a `Plan` of type `schema:creator`.

### CODE REVIEW

File: renku-model/src/test/scala/io/renku/graph/model/entities/ActivityLensSpec.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import io.renku.graph.model.RenkuUrl
+import io.renku.graph.model.projects.DateCreated
+import io.renku.graph.model.testentities.generators.{ActivityGenerators, EntitiesGenerators}
+import org.scalatest.matchers.should
+import org.scalatest.wordspec.AnyWordSpec
+
+import java.time.Instant
+
+class ActivityLensSpec extends AnyWordSpec with should.Matchers {
```
Comment (jachro) [inline renku-model/src/test/scala/io/renku/graph/model/entities/ActivityLensSpec.scala]: I think, if you'd import `io.renku.graph.model.testentities._` you could remove the implicit `renkuUrl` val as well as just use the `activityEntities`, `planEntities` and other without necessity of reaching to relevant generator classes.

Comment (eikek) [reply renku-model/src/test/scala/io/renku/graph/model/entities/ActivityLensSpec.scala]: Thanks! 👍🏼

File: renku-model/src/test/scala/io/renku/graph/model/entities/ActivityLensSpec.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      val person   = createActivity.author
+      ActivityLens.activityAuthor.get(activity)         shouldBe activity.author
+      ActivityLens.activityAuthor.set(person)(activity) shouldBe activity.copy(author = person)
+    }
+  }
+
+  private def createActivity =
+    ActivityGenerators
+      .activityEntities(ActivityGenerators.planEntities()(ActivityGenerators.planCommands))
+      .apply(DateCreated(Instant.EPOCH))
```
Comment (jachro) [inline renku-model/src/test/scala/io/renku/graph/model/entities/ActivityLensSpec.scala]: It's absolutely fine to fix the DateCreated on Epoch but as there are generators for everything, you could also import `GraphModelGenerators.projectCreatedDates` and generate one.

File: renku-model/src/test/scala/io/renku/graph/model/entities/AssociationLensSpec.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      val assoc4 = createAssociationPerson
+      AssociationLens.associationAgent.set(assoc1.agent.asLeft)(assoc3) shouldBe
+        assoc3.copy(agent = assoc1.agent)
+      AssociationLens.associationAgent.set(assoc2.agent.asRight)(assoc4) shouldBe
+        assoc4.copy(agent = assoc2.agent)
+    }
+  }
+
+  private def createPlan =
+    EntitiesGenerators
```
Comment (jachro) [inline renku-model/src/test/scala/io/renku/graph/model/entities/AssociationLensSpec.scala]: We can eventually take more or less similar changes as I described for the `ActivitiesLensSpec` if we like.

File: renku-model/src/test/scala/io/renku/graph/model/entities/PlanLensSpec.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      val p1 = createPlan
+      val p2 = createPlan
+
+      PlanLens.planCreators.get(p1)              shouldBe p1.creators
+      PlanLens.planCreators.set(p2.creators)(p1) shouldBe p1.copy(creators = p2.creators)
+    }
+  }
+
+  private def createPlan =
+    EntitiesGenerators
```
Comment (jachro) [inline renku-model/src/test/scala/io/renku/graph/model/entities/PlanLensSpec.scala]: Eventually, we can do what I described for `ActivitiesLensSpec` if we like.

File: renku-model/src/test/scala/io/renku/graph/model/entities/ProjectSpec.scala [hunk 5]
```diff
@@ -524,14 +597,16 @@ class ProjectSpec extends AnyWordSpec with should.Matchers with ScalaCheckProper
...
-                                 cliDate,
-                                 activities = Nil,
-                                 datasets = Nil
+      val jsonLD = cliLikeJsonLD(
+        resourceId,
+        cliVersion,
+        schemaVersion,
+        projectInfo.maybeDescription,
+        projectInfo.keywords,
+        None,
```
Comment (jachro) [inline renku-model/src/test/scala/io/renku/graph/model/entities/ProjectSpec.scala:606]: Maybe we can name the parameter so we know what's the `None` for.

File: triples-generator/src/test/scala/io/renku/triplesgenerator/events/consumers/tsprovisioning/transformation/namedgraphs/activities/KGInfoFinderSpec.scala [hunk 6]
```diff
@@ -57,7 +58,7 @@ class KGInfoFinderSpec
       val person = personEntities.generateOne.to[entities.Person]
       upload(
         to = projectsDataset, {
-          implicit val enc: JsonLDEncoder[entities.Person] =
+          implicit val encß: JsonLDEncoder[entities.Person] =
```
Comment (jachro) [inline triples-generator/src/test/scala/io/renku/triplesgenerator/events/consumers/tsprovisioning/transformation/namedgraphs/activities/KGInfoFinderSpec.scala]: Is that a typo in the enc val name?

**Review by jachro — APPROVED** [review]
Thanks Eike! The changes look really good! I've learned about Lens compose which is pretty nice!

File: triples-generator/src/test/scala/io/renku/triplesgenerator/events/consumers/tsprovisioning/transformation/namedgraphs/activities/KGInfoFinderSpec.scala (hunk 6 above)
Comment (eikek) [reply triples-generator/src/test/scala/io/renku/triplesgenerator/events/consumers/tsprovisioning/transformation/namedgraphs/activities/KGInfoFinderSpec.scala]: Oops, yes :) thanks!

**Pull request merged**